### PR TITLE
STB inclusion workarounds for Debian buster

### PR DESCRIPTION
**Pull request by rcurtin**
This follows up to a discussion in IRC that @zoq pointed out, where there are problems with STB on Debian buster due to some missing `static` markings on functions in STB, fixed by this commit:

https://github.com/nothings/stb/commit/feb9de355d52eb90abda31cd6883d944e6fe06ca#diff-469d4cb48b9965801e6c8cb7b92e46fe

The PR here is a workaround, to make sure that the implementations are only included once in the entire mlpack codebase via the `STB_IMAGE_IMPLEMENTATION` macro.  I also removed unnecessary includes from the data loading code to simplify it.

### CODE REVIEW

**Comment by birm** [comment]
I don't think the CI builds will return with a merge conflict.

**Comment by rcurtin** [comment]
Good point, just merged in master now.  Hopefully that will help. :)

File: src/mlpack/tests/main_tests/preprocess_encoding_string_test.cpp [hunk 1]
```diff
@@ -0,0 +1,411 @@
+/**
+ * @file preprocess_encoding_string_test.cpp
```
Comment (birm) [inline src/mlpack/tests/main_tests/preprocess_encoding_string_test.cpp:2]: Why is this even in the diff?

Comment (rcurtin) [reply src/mlpack/tests/main_tests/preprocess_encoding_string_test.cpp:2]: Oh, wow, I kind of dropped the ball on this one.  It looks like I built my branch on top of the string encoding PR.  Let me redo this PR...

**Pull request closed without merging**